lib/galaxy/container/docker_swarm_manager.py

```diff
@@ -0,0 +1,560 @@
+"""
+Docker Swarm mode management
+"""
+from __future__ import absolute_import
+
+import argparse
+import errno
+import logging
+import os
+import subprocess
+import sys
+import time
+
+try:
+    import daemon
+    import daemon.pidfile
+    import lockfile
+except ImportError:
+    daemon = None
+
+try:
+    import galaxy   # noqa: F401 this is a test import
+except ImportError:
+    sys.path.insert(0, os.path.abspath(os.path.join(
+        os.path.dirname(__file__),
+        os.pardir,
+        os.pardir)))
+
+from galaxy.config import (
+    configure_logging,
+    find_path,
+    find_root,
+)
```

Comment:
It seems like there are good abstractions here I'd like to leverage in Pulsar and Planemo. I'd like to just move this all into ``galaxy-lib`` but this script can't be because of these ``galaxy.config`` things.  I could put just a subset of the module in of course - but I usually take this as a sign that a potential library/utility package is attempting to access too much of Galaxy. Maybe it makes some sense to move this one script into ``galaxy.webapps.galaxy`` or keep this script here and move everything else into ``galaxy.util.containers``.